text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
```

Comment:
@aturon I think what @mahkoh meant was that we could potentially allocate just once and return &'static str / OsStr into that "hidden" global allocation.

This would let calling os::args() multiple times only allocate once, which is a nice win for usability (no need to save them off somewhere for short scripts, etc to avoid allocating every timie)